velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -70,22 +244,151 @@ std::unique_ptr<velox::dwrf::Writer> HiveDataSink::createWriter() {
   // Without explicitly setting flush policy, the default memory based flush
   // policy is used.
 
-  auto hiveWriterParameters =
-      std::dynamic_pointer_cast<const HiveWriterParameters>(
-          writeProtocol_->getWriterParameters(
-              insertTableHandle_, connectorQueryCtx_));
-  VELOX_CHECK_NOT_NULL(
-      hiveWriterParameters,
-      "Hive data sink expects write parameters for Hive.");
-  writerParameters_.emplace_back(hiveWriterParameters);
-
-  auto writePath = fs::path(hiveWriterParameters->writeDirectory()) /
-      hiveWriterParameters->writeFileName();
+  auto writePath = fs::path(writerParameters->writeDirectory()) /
+      writerParameters->writeFileName();
   auto sink = dwio::common::DataSink::create(writePath);
   return std::make_unique<Writer>(
       options, std::move(sink), *connectorQueryCtx_->memoryPool());
 }
 
+void HiveDataSink::moveWriters() {
+  raw_vector<uint64_t> newIndices;
+
+  SelectivityVector validRows = SelectivityVector::empty(writers_.size());
+  for (vector_size_t i = 0; i < writers_.size(); i++) {
+    if (writers_[i]) {
+      validRows.setValid(i, true);
+    }
+  }
+  validRows.updateBounds();
+
+  // All valid values of partitionsVector_ are from previous addInput() that
+  // have been fed into partitionIdGenerator at least once, and thus assumed not
+  // to cause partitionIdGenerator to rehash.
+  partitionIdGenerator_.stableRun(partitionsVector_, validRows, newIndices);
```

Comment:
Nope, validRows has a subset of rows selected. Not all rows of partitionsVector_ are selected since the index of partitionsVector_ generated from VectorHasher are not consecutive.